app/javascript/components/widgets/weather/sunrise-sunset.jsx

```diff
@@ -53,62 +53,60 @@ const SunsetLabel = styled.div`
 const getSunriseSunsetLocation = gql`
   fragment SunriseSunsetLocation on Location {
     timezone
-  }
-`;
-
-const getSunriseSunsetWeather = gql`
-  fragment SunriseSunsetWeather on Weather {
-    daily {
-      data {
-        sunriseTime
-        sunsetTime
-      }
+    solarcycles {
+      type
+      time
     }
   }
 `;
 
-const SunriseSunset = ({ weather, location }) => {
-  const { sunriseTime, sunsetTime } = weather.daily.data[0];
-  const { timezone } = location;
+const SunriseSunset = ({ location }) => {
+  const { timezone, solarcycles } = location;
+  const currDate = new Date();
+
+  solarcycles.forEach(element => {
+    console.log(element);
+  });
+  const startTime = solarcycles
+    .filter(cycle => new Date(cycle.time).getTime() - currDate.getTime() < 0)
+    .slice(-1)[0];
```

Comment:
Pre-empting @cbarber with `takeLast`?

Reply:
Close! I think you can probably name things a little better:
```
const [beforeNow, afterNow] = slitWhen(cycle => ....).;
const startTime = takeLast(1, afterNow);
const stopTime = take(1,afterNow), 
```

This includes a pet peeve of mine. Always start/stop or begin/end. Never start/end or begin/stop :)